ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/cache/HttpCacheEntry.kt

```diff
@@ -92,13 +97,21 @@ internal fun HttpResponse.cacheExpires(fallback: () -> GMTDate = { GMTDate() }):
 
 internal fun HttpCacheEntry.shouldValidate(request: HttpRequestBuilder): ValidateStatus {
     val cacheControl = parseHeaderValue(responseHeaders[HttpHeaders.CacheControl])
+    val requestCacheControl = parseHeaderValue(request.headers[HttpHeaders.CacheControl])
+
+    if (CacheControl.NO_CACHE in requestCacheControl) return ValidateStatus.ShouldValidate
+
+    val requestMaxAge = requestCacheControl.firstOrNull { it.value.startsWith("max-age=") }
```

Comment:
Could you tell me if `max-age` can go second? Please consider adding unit test

Reply:
`requestCacheControl` is parsed header value